packages/core/src/messages/vector-index-filters.ts

```diff
@@ -0,0 +1,194 @@
+/**
+ * Base class for all vector index filters.
+ */
+export abstract class VectorFilterExpression {
+  public abstract toString(): string;
+
+  /**
+   * Creates an AND expression between this expression and another one.
+   * @param other The other expression.
+   * @returns The AND expression.
+   */
+  public and(other: VectorFilterExpression): VectorFilterAndExpression {
+    return VectorFilterExpressionFactory.and(this, other);
+  }
+
+  /**
+   * Creates an OR expression between this expression and another one.
+   * @param other The other expression.
+   * @returns The OR expression.
+   */
+  public or(other: VectorFilterExpression): VectorFilterOrExpression {
+    return VectorFilterExpressionFactory.or(this, other);
+  }
+
+  /**
+   * Negates this expression.
+   * @returns The negated expression.
+   */
+  public not(): VectorFilterNotExpression {
+    return VectorFilterExpressionFactory.not(this);
+  }
+}
+
+/**
+ * Factory for creating vector filter expressions.
+ */
+export class VectorFilterExpressionFactory {
```

Comment:
I added this factory class to ease the pain building a nontrivial expression. The hard way (long names and lots of `new`s):

```typescript
new VectorFilterAndExpression(new VectorFilterEqualsExpression('field1', 'value1'), new VectorFilterEqualsExpression('field2', 42))
```

And with these conveniences:
```typescript
import { VectorFilterExpressionFactory as F } from package;

F.equals('field1', 'value1').and(F.equals('field2', 42));
// ...or...
F.and(F.equals('field1', 'value1'), F.equals('field2', 42));
```

Reply:
I'm 💯 on having this factory; I think in modern languages/stdlibs, I've more frequently seen stuff like this exposed by just adding an `s` on the name of the thing, so like we did with `Configurations`, I'd probably name this `VectorFilterExpressions`.

Reply:
Done in [c4aeb19](https://github.com/momentohq/client-sdk-javascript/pull/1119/commits/c4aeb19f4b45d8c0343a7f98277c64dc9cfb058c)